File: 백준/Bronze/2446. 별 찍기 － 9/별 찍기 － 9.py
import sys
n=int(sys.stdin.readline())
end=2*n-1
med=end//2
for i in range(end):
    if i<=med:
        for j in range(i):
            print(' ',end='')
        for j in range(i,end-i):
            print('*',end='')
    else: # i>end//2
        for j in range(med*2-i):
            print(' ',end='')
        for j in range(med*2-i,end-(med*2-i)):
            print('*',end='')
    print()

File: 백준/Bronze/2446. 별 찍기 － 9/별 찍기 － 9.c
#include <stdio.h>
int main(void) {
    int n, end, med;
    scanf("%d", &n);
    end = 2 * n - 1;
    med = end / 2;
    for (int i = 0; i < end; i++) {
        if (i <= med) {
            for (int j = 0; j < i; j++)
                printf(" ");
            for (int j = i; j < end - i; j++)
                printf("*");
        } else {
            for (int j = 0; j < med * 2 - i; j++)
                printf(" ");
            for (int j = med * 2 - i; j < end - (med * 2 - i); j++)
                printf("*");
        }
        printf("\n");
    }
    return 0;
}